lib/src/layout/wallpaper_tinted_area.dart

```diff
@@ -0,0 +1,141 @@
+          backgroundColor: backgroundColor,
+          child: child,
+        ),
+      );
+    }
+
+    return _WallpaperTintedAreaLayoutBuilder(
+      backgroundColor: backgroundColor,
+      child: child,
+    );
+  }
+}
+
+class _WallpaperTintedAreaLayoutBuilder extends StatelessWidget {
+  const _WallpaperTintedAreaLayoutBuilder({
+    required this.backgroundColor,
+    required this.child,
+  });
+
+  /// The color to apply to the background when wallpaper tinting is disabled.
+  final Color backgroundColor;
+
+  /// The widget below this widget in the tree.
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    if (GlobalWallpaperTintingSettings
+        .data.isWallpaperTintingDisabledByWindow) {
+      return Container(
+        decoration: BoxDecoration(
+          color: backgroundColor,
+        ),
+        child: child,
+      );
+    }
+
+    // This LayoutBuilder forces the widget to be rebuilt when a layout change
+    // is detected. This is necessary for the VisualEffectSubviewContainer to
+    // be updated.
+    return LayoutBuilder(
+      builder: (context, _) {
+        return VisualEffectSubviewContainer(
+          material: NSVisualEffectViewMaterial.windowBackground,
+          child: WallpaperTintingSettingsBuilder(
+            builder: (context, data) {
+              final isWallpaperTintingEnabled = data.isWallpaperTintingEnabled;
+
+              return _WallpaperTintedAreaTweenAnimationBuilder(
+                isWallpaperTintingEnabled: isWallpaperTintingEnabled,
+                backgroundColor: backgroundColor,
+                child: child,
+              );
+            },
+          ),
+        );
+      },
+    );
+  }
+}
+
+class _WallpaperTintedAreaTweenAnimationBuilder extends StatelessWidget {
+  const _WallpaperTintedAreaTweenAnimationBuilder({
+    required this.isWallpaperTintingEnabled,
+    required this.backgroundColor,
+    required this.child,
+  });
+
+  /// Whether wallpaper tinting is enabled.
+  final bool isWallpaperTintingEnabled;
+
+  /// The color to apply to the background when wallpaper tinting is disabled.
+  final Color backgroundColor;
+
+  /// The widget below this widget in the tree.
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    return TweenAnimationBuilder<double>(
+      duration: const Duration(milliseconds: 100),
+      tween: Tween<double>(
+        begin: isWallpaperTintingEnabled ? 0.0 : 1.0,
+        end: isWallpaperTintingEnabled ? 0.0 : 1.0,
+      ),
+      builder: (context, value, child) {
+        return Container(
+          decoration: BoxDecoration(
+            color: backgroundColor.withOpacity(value),
+            backgroundBlendMode: BlendMode.src,
+          ),
+          child: child,
+        );
+      },
+      child: Opacity(
+        // For some reason, omitting this Opacity widget causes
+        // a dark background to appear.
+        opacity: 1.0,
+        child: child,
+      ),
```

Comment:
It’s been a while since I wrote this, but I believe the background was caused by the Cupertino scaffold. The `backgroundBlendMode: BlendMode.src` attribute of the `Container` inside the builder should, theoretically, remove that background, but for some reason, it doesn’t unless the `Opacity` widget is present. I’ll try to investigate further.

Reply:
Turns out that the opacity widget could be replaced with a repaint boundary. I assume `Opacity` just used a `RepaintBoundary` under the hood which allowed the `BlendMode.src` to work correctly.